### PR TITLE
add API handler to forward payload to Segment

### LINKED ISSUE #279: API route for Segment integration
We need to implement an API route that facilitates forwarding the payload sent by the Telemetry Service to the Segment API. This integration will allow us to track and analyze telemetry data effectively.

We can create a new API endpoint in the documentation app to receive the telemetry payload.

### PR DESCRIPTION
**Pull request by eliobricenov**
closes #279

### CODE REVIEW

File: docs/pages/api/telemetry/index.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { Analytics } from "@segment/analytics-node";
+import { NextApiRequest, NextApiResponse } from "next";
+
+export default async function forwardTelemetry(
```
Comment (0xOlias) [inline docs/pages/api/telemetry/index.ts:4]: Do we want this to be an edge function? Genuinely not sure! (https://nextjs.org/docs/pages/building-your-application/routing/api-routes#edge-api-routes)

File: docs/pages/api/telemetry/index.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    console.error('Missing required environment variable "SEGMENT_WRITE_KEY".');
+    return res
+      .status(500)
+      .json({ error: "An error occurred while processing telemetry data." });
+  }
+
+  try {
+    const analytics = new Analytics({
+      writeKey: process.env.SEGMENT_WRITE_KEY,
+    });
```
Comment (0xOlias) [inline docs/pages/api/telemetry/index.ts:23]: I wonder if we can do this stuff above, outside the function scope.

File: docs/pages/api/telemetry/index.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    await analytics.track({
+      ...req.body,
+    });
+
+    res.status(200).json({
+      message: "Telemetry data processed successfully.",
+    });
```
Comment (0xOlias) [inline docs/pages/api/telemetry/index.ts:31]: Do we need to await the request to Segment? I wonder if we can actually send the HTTP response back to the client immediately, then try to forward to segment.

**Review by 0xOlias — APPROVED** [review]
LGTM! Couple ideas/questions

File: docs/pages/api/telemetry/index.ts (hunk 3 above)
Comment (eliobricenov) [reply docs/pages/api/telemetry/index.ts:31]: we can do that - the tradeoff is that the client will never know if the event was forwarded successfully. In this case, I don't think it is important but it'd be good to track if the forwarding fails

File: docs/pages/api/telemetry/index.ts (hunk 2 above)
Comment (eliobricenov) [reply docs/pages/api/telemetry/index.ts:23]: no :/ - I tried doing it outside of the function but Typescript complains.

this is what chatgpt says (although I'm not exactly sure this is true):

> In Node.js, environment variables are usually set when the process starts and are available for the duration of the process. However, in a serverless or "edge" environment (like the one you're running your Next.js code in), individual requests are handled in isolation and might not share the same environment variables state.

File: docs/pages/api/telemetry/index.ts (hunk 1 above)
Comment (eliobricenov) [reply docs/pages/api/telemetry/index.ts:4]: i'm not sure either but it seems to work!

File: docs/pages/api/telemetry/index.ts (hunk 3 above)
Comment (0xOlias) [reply docs/pages/api/telemetry/index.ts]: Yea. I wonder if Vercel has a dashboard for "failed edge function invocations" or something. Like, maybe we can just throw in the function body if the `analytics.track` call fails. In any case, this works as is for now.

I don't think the client should know/care if the forwarding failed - there's nothing they can do about it.

**Review by 0xOlias — APPROVED** [review]
LGTM!

**Pull request merged**